### DISCUSSION TITLE
Best way to support a new keyboard in QMK? (cantor)

### DISCUSSION THREAD

**Discussion by nilokr** [manna-harbour/miryoku#89] [Q&A] [answered]
Hi! I have made a 36-key version of the [cantor](https://github.com/diepala/cantor) keyboard and I would like to know the best way to support it on miryoku QMK. I couldn't find documentation on that (at least something that I would understand at this point 😬), so any pointers would be greatly appreciated. Thanks!

**Comment by manna-harbour** [comment, marked as answer]
If your board is equivalent to a cantor but with the outer pinkie columns removed (i.e. it uses the same schematic and can run the same firmware) then you don't need to do anything, as cantor is already supported (see below) and the outer columns are not used.

If you used a different schematic then you only need to enable the `split_3x5_3` community layout in your keyboard definition, which is already supported by Miryoku QMK.

* https://github.com/qmk/qmk_firmware/blob/master/docs/feature_layouts.md
* use `LAYOUT_split_3x5_3` as the layout macro name and add `LAYOUTS = split_3x5_3` to `rules.mk` at the keyboard level.
* https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#split_3x5_3

Cantor is supported in Miryoku QMK via the `split_3x6_3` community layout, but cantor is not in the `miryoku` branch and current Miryoku QMK is not in upstream `master`. You can build for cantor now from QMK `master` which includes an older version of Miryoku,  or by merging the `miryoku` branch with upstream `master` as in this run https://github.com/manna-harbour/miryoku_qmk/actions/runs/2401019701 using the merge feature https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#workflow-builds. Also the `miryoku` branch will soon be rebased on upstream `master`, and there is an open PR with QMK, so after that you'll be able to build current Miryoku QMK for cantor directly from `miryoku` or upstream `master`.

**Comment by nilokr** [comment]
Thank you! I had no idea cantor was already supported (my version has the 3 outer keys removed just as you said). All I had to do was `qmk flash -kb cantor -km manna-harbour_miryoku`. Merging your PR branch locally before flashing also worked. I really appreciate your work, thanks again for putting the effort to develop and share Miryoku, and for helping the community.
